### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by NanakiC** [gorgen2020/DVGNN#1]
Dear author, I noticed that the dimension of T_dtrive is (1766, 64), while the adjacency matrix generated by GCN from T_dtrive has a dimension of (1765, 64). It seems that the adjacency matrix was generated using all the data, not just the training set. Could this potentially lead to data leakage?

**Comment by gorgen2020** [comment]
Hello NanakiC,
Thank you very much for your interest in our research and for pointing this out.
You're correct that during the graph generation stage, we input the entire T_drive dataset (with shape (1766, 64)) into the model to generate the adjacency matrix. However, we would like to clarify that only the training subset of the data is used for training, while the test data is reserved strictly for inference.
If you take a look at Tdrive.conf, you’ll notice the following configuration:
##########
train_rate = 0.8
edges_test_ratio = 0.2
###########
These are parsed in model.py as:
##################
self.train_rate = float(training_config['train_rate'])
self.test_ratio = float(training_config['edges_test_ratio'])
#################
During training (Train() function in model.py), the adjacency matrix is split using the configured ratios:
#################
train_coo_adjacency, test_edges, test_edges_neg, valid_edges, valid_edges_neg = GraphReader.SplitTrainTestDataset(
    coo_adjacency, self.test_ratio, self.valid_ratio)
#################
Then in training:
#############################
for **m** in range(int(**self.train_rate** * trainX.shape[1])):
                if self.method == 'dynamic':
                    feed_dict = {self.tf_sparse_adjacency: (edges, values),
                                     self.tf_norm_sparse_adjacency: (norm_edges, norm_values), self.inputs1: np.reshape(**trainX[:, m:m+self.seq_len]**, [self.n_nodes, self.seq_len * self.features_num]),
                                     self.inputs2: np.reshape(**trainX[:, m + 1:m + 1 + self.seq_len],**[self.n_nodes, self.seq_len * self.features_num])}
##############################
Only the training portion is used. To summarize: although the initial graph is generated using the entire dataset, the test data is excluded from the training process. The trained model is then used to make predictions on the test data. Thus, no data leakage occurs, since the test set is not used in training—neither during graph generation training nor in the final prediction stage.

Thanks again for your valuable feedback!

**Comment by NanakiC** [comment]
Thank you very much for your answer！
